src/gui/shell.cpp

```diff
@@ -867,6 +866,28 @@ void Shell::handleGuiFontFunction(const QVariantList& args)
 	setGuiFont(fdesc, force, true /*setOption*/);
 }
 
+void Shell::handleCloseEvent(const QVariantList &args)
+{
+	if (args.size() == 2 && !args.at(1).canConvert<int>()) {
+		qWarning() << "Unexpected exit status for close:" << args.at(1);
+		return;
+	} else if (args.size() >= 3) {
+		qWarning() << "Unexpected arguments for close:" << args;
+		return;
+	}
+
+	qDebug() << "Neovim requested a GUI close";
```

Comment:
Let's get rid of the `>=3` case, and change to `>=2`. There could be additional parameters in the future...

``` C++
qDebug() << "Neovim requested a GUI close";

if (args.size() >= 2 && !args.at(1).canConvert<int>()) {
	qWarning() << "Unexpected exit status for close:" << args.at(1);
	return;
}
```

Might as well add the qDebug at the very top. The request was made even if malformed.